### DISCUSSION TITLE
Warning with optional arguments with latest pyo3 version 0.22

### DISCUSSION THREAD

**Discussion by bam5wi** [PyO3/pyo3#4289] [Questions]
With the latest pyo3 update optional arguments now produce a warning. With a function signature the warning can be suppressed. This is possible for pyfunctions, but not for setter, because for setters it is not allowed to use the signature feature.

How can I set a default value for an optional argument in a setter?

The following example fails to compile:

```rust
    #[setter]
    #[pyo3(signature = (pretty,))]
    fn set_bool_true(&mut self, value: Option<String>) {
        self.0.bool_true = value.map(|v| v.into_bytes());
    }
```

It should be possible to provide None as argument. How can I solve this?

**Comment by davidhewitt** [comment]
This is unfortunate. I think that in this case the warning shouldn't fire, as you can only reach the setter through `myclass.bool_true = None` anyway, and there's no possibility of using a default value for assignment in this way.

**Reply by davidhewitt** [reply]
NB we opened this as an issue https://github.com/PyO3/pyo3/issues/4292 and fixed it in 0.22.1.
